### ISSUE TITLE
Can't get products. Return empty array. How to resolve?

### ISSUE THREAD

**Issue by rubinchyk** [AlexDisler/cordova-plugin-inapppurchase#80]
Chronology of what I did: 
1) Set: 
`cordova plugin add cc.fovea.cordova.purchase --variable BILLING_KEY = " My key " `

2) add to the manifest.json:
`{" play_store_key ":" MyRSA key "} `

3) AndroidManifest added permission for billing:
```
<uses-permission android:name="android.permission.BILLING" />
<uses-permission android:name="com.android.vending.BILLING" />
```

4) In the Template page I have button with loading products and field with foreach of product list.

5) The controller file:

```
$scope.loadProducts = function () {
      $ionicLoading.show({ template: spinner + 'Loading Products...' });

      alert('If Isset inAppPurchase: ' + inAppPurchase);

      inAppPurchase
      .getProducts(productIds)
      .then(function (products) {
        $ionicLoading.hide();
        alert('If Isset products: ' + products);
        $scope.products = products;
      })
      .catch(function (err) {
        $ionicLoading.hide();
        alert('Error: ' + err);
        alert(err);
      });
    };
```

6) Create a build: 
`cordova build --release android`
, sign, archive and install to a real phone. 

7) On Google Play have the same version of the android - beta, status Published. 2 Products in Active status.

Q: What have I forgotten? I did not finish? By clicking on the button to receive all of the products, the result - an empty array on success of promise. I would appreciate any comments!

**Comment by jashick** [comment]
Is the BILLING_KEY used in step 1 the same as the RSA key in step 2?  If not, where did you find the Billing Key?

**Comment by rubinchyk** [comment]
@jashick Yes the same, I'm using billing key from 'Services & API' from Google Play (Base64-encoded RSA public key).

**Comment by jashick** [comment]
@rubinchyk I am having the same problem as you then.  I continue to get an empty array when calling getProducts

**Comment by jashick** [comment]
After much trial and error, I was able to get it working.  In the end, I had to upload a signed APK to the play store as an alpha release.  The version in alpha has to match the version on the testing device.  Also, I had to publish the app.  If it is in Draft it won't work.  It had to be published.  Since I am still testing and haven't uploaded a Production APK, it isn't live on the app store yet, even though it shows as published.

**Comment by anonrig** [comment]
You're installing a different plugin rather than the plugin you're writing the isssue for.

**Comment by rubinchyk** [comment]
Yes, you right, it's my text mistake. I rewrite to correct plugin which I have. Problem still relevant. 
App consist the same version like in Google App store in Beta and Published.

**Comment by rubinchyk** [comment]
@jashick did you use code from example?

**Comment by vernondegoede** [comment]
I'm having this issue as well on iOS. `getProducts` returns an empty array of products.

**Comment by jashick** [comment]
@rubinchyk No, I didn't use the code above.  My issue ended up being that although I had uploaded an APK to alpha, I had not published my app.  It was still listed as draft.  Setting it to published resolved my issue.

**Comment by Jay031** [comment]
@rubinchyk @vernondegoede I have the same problem; 
As @vernondegoede  already said: `getProducts `returns an empty array of products on iOS.

Do you guys have any solution for this already? I've been working on it for days now

**Comment by vernondegoede** [comment]
I haven't found a solution for this yet. This plugin seems to be using the RMStore, which hasn't been updated for a couple of months already. 

Our client decided to skip in app purchases for now since there doesn't seem to be a solution.

**Comment by JohnScottMcNerney** [comment]
@rubinchyk @vernondegoede
Ditto on iOS returning an empty array.

**Comment by vernondegoede** [comment]
@JohnScottMcNerney Do you have a valid Paid contract in iTunes Connect? This appeared to be the problem at our project.

**Comment by smallbiz13** [comment]
@vernondegoede we have a valid paid contract in iTunes Connect as well. But still returning an empty array.

**Comment by kcedric** [comment]
had the same problem : for android apps you need to deploy a signed apk on your device to test in app purchases.

So basically, either you deploy a beta on goole play store and install it on the device or you generate a signed apk and install it on the device by using "adb install".

**Comment by smallbiz13** [comment]
Turns out that the Paid Contract Agreement was not completed properly. Works now. Thanks all.

**Comment by RedekProject** [comment]
Good evening everyone !

I post this message to tell you that I managed to solve the problem of the empty array. Indeed I work there for more than 6 hours on and it turns out that my problem came from the "productsId". Indeed we can see everywhere that we have to put rather `com.nameapp.productName` while we just have to put `productName`.

Hope you helped!

**Comment by vernondegoede** [comment]
Maybe we should add some `known issues` section to the Readme to prevent others from experiencing the same issues?

**Comment by RedekProject** [comment]
I proposed a `pull request` of the ReadMe with modification and add new section `known issues`. It is awaiting validation by the creator.
